apis/Google.Datastore.V1Beta3/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs

```diff
@@ -90,10 +91,11 @@ public void StructuredQuery()
             {
                 Filter = Filter.Equal("author", "Jane Austen")
             };
-            DatastoreQueryResults results = db.RunQuery(query);
-            // DatastoreQueryResults implements IEnumerable<Entity>, but you can
-            // call AsEntityResults(), AsBatches() or AsResponses() to see the query
-            // results in whatever way makes most sense for your application.
+            LazyDatastoreQuery results = db.RunQuery(query);
```

Comment:
jskeet, I really like this API your proposing, with one minor tweak:
```
DatastoreQueryResults RunQuery(...)
LazyDatastoreQuery RunQueryLazily(...)
```

The `Eagerly` suffix isn't necessary.  When ever I see a function DoFoo(), I assume it does it immediately (eagerly).